.github/workflows/integration_tests.yml

```diff
@@ -248,3 +383,54 @@ jobs:
           if [[ "${{ job.status }}" != "success" ]]; then
             exit 1
           fi
+
+  add_success_label:
```

Comment:
It seems odd that the check failure above is part of the tests job, but the two label operations below are their own job.  Could the blow jobs be added as steps to the above job, just like the add failure label and comment steps?  They all could share the same get time, a step which then wouldn't need any conditionals.

Reply:
The failure is part of the tests job because we mark failure if even a single matrix entry fails. Whereas we mark success only once the entire matrix is complete, if there were no failures..

Reply:
I originally had them combined but it actually didn't quite work - using success() / failure() to check the status of the entire workflow only only works in an "if:" at the job level, if you call them in an "if:" at the steps level, it only checks if the *current* job succeeded or failed. So the triggers need to be jobs-based.

Reply:
Hm, interesting.